### PR TITLE
Custom detector name

### PR DESCRIPTION
**Pull request by zricethezav**
Adds `name` to the extra data map for custom detectors.

Ex output:

```
Found unverified result 🐷🔑❓
Detector Type: CustomRegex
Decoder Type: PLAIN
Raw result: eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
Name: Super custom API detector
File: /Users/zrice/code/trufflesecurity/sandbox/config-demo/test.md
Line: 1
```

Takes care of https://github.com/trufflesecurity/trufflehog/issues/1330 and https://github.com/trufflesecurity/trufflehog/issues/1063

### CODE REVIEW

File: pkg/custom_detectors/custom_detectors.go [hunk 1]
```diff
@@ -99,6 +99,10 @@ func (c *customRegexWebhook) FromData(ctx context.Context, verify bool, data []b
 	close(resultsCh)
 
 	for result := range resultsCh {
+		// NOTE: I don't believe this is being set anywhere else, hence the map assignment.
+		result.ExtraData = map[string]string{
```
Comment (zricethezav) [inline pkg/custom_detectors/custom_detectors.go:103]: for the reviewer: could check for `nil` here but it doesn't look as nice. dont have a strong opinion on this one.

**Review by mcastorina — COMMENTED** [review]
We are also assigning `c.GetName()` to the `Result.DetectorName` on line 120. Are we not displaying that information anywhere?

**Comment by mcastorina** [comment]
Should we add the entry there instead of including it in `ExtraData`? Maybe with a check to prevent outputting non-empty names?

I have a vague recollection that `DetectorName` was added for this use case.

**Comment by zricethezav** [comment]
> Should we add the entry there instead of including it in ExtraData? Maybe with a check to prevent outputting non-empty names? 

I have a small pushback on that since `DetectorName` is only used for CustomRegex detectors which means we would have to introduce some extra logic to check if DetectorName is set in `PrintPlainOutput`. Not a huge deal but imo a little less clean than relying on the metadata. Happy to discuss further if you feel strongly on this

**Review by mcastorina — APPROVED** [review]
No strong opinion either way, but maybe we should remove `DetectorName` if we aren't using it.

**Pull request merged**